src/software/geom/angle_map_test.cpp

```diff
@@ -0,0 +1,96 @@
+#include "software/geom/angle_map.h"
+
+#include <gtest/gtest.h>
+
+TEST(AngleMapTest, add_obstacle_angle_segment_takes_entire_angle_map)
+{
+    Angle top_angle    = Angle::half();
+    Angle bottom_angle = Angle::zero();
+    AngleMap angle_map = AngleMap(top_angle, bottom_angle, 1);
+
+    AngleSegment obstacle_angle_seg = AngleSegment(top_angle, bottom_angle);
+    angle_map.addNonViableAngleSegment(obstacle_angle_seg);
+
+    EXPECT_EQ(0, angle_map.getBiggestViableAngleSegment().getDelta());
+}
+
+TEST(AngleMapTest, add_obstacle_angle_segment_takes_lower_half_angle_map)
+{
+    Angle top_angle    = Angle::half();
+    Angle bottom_angle = Angle::zero();
+    AngleMap angle_map = AngleMap(top_angle, bottom_angle, 1);
+
+    AngleSegment obstacle_angle_seg = AngleSegment(Angle::quarter(), Angle::zero());
+    angle_map.addNonViableAngleSegment(obstacle_angle_seg);
+
+    EXPECT_EQ(90, angle_map.getBiggestViableAngleSegment().getDelta());
+}
+
+TEST(AngleMapTest, add_obstacle_angle_segment_takes_upper_half_angle_map)
+{
+    Angle top_angle    = Angle::half();
+    Angle bottom_angle = Angle::zero();
+    AngleMap angle_map = AngleMap(top_angle, bottom_angle, 1);
+
+    AngleSegment obstacle_angle_seg = AngleSegment(Angle::half(), Angle::quarter());
+    angle_map.addNonViableAngleSegment(obstacle_angle_seg);
+
+    EXPECT_EQ(90, angle_map.getBiggestViableAngleSegment().getDelta());
+}
+
+TEST(AngleMapTest, add_obstacle_angle_segment_contained_within_another)
+{
+    Angle top_angle    = Angle::half();
+    Angle bottom_angle = Angle::zero();
+    AngleMap angle_map = AngleMap(top_angle, bottom_angle, 1);
+
+    AngleSegment obstacle_angle_seg = AngleSegment(Angle::quarter(), Angle::zero());
+    angle_map.addNonViableAngleSegment(obstacle_angle_seg);
+
+    AngleSegment overlapping_angle_seg =
+        AngleSegment(Angle::fromDegrees(45), Angle::zero());
+    angle_map.addNonViableAngleSegment(overlapping_angle_seg);
+
+    EXPECT_EQ(90, angle_map.getBiggestViableAngleSegment().getDelta());
+}
+
+TEST(AngleMapTest, add_obstacle_angle_segment_overlaps_with_another)
+{
+    Angle top_angle    = Angle::half();
+    Angle bottom_angle = Angle::zero();
+    AngleMap angle_map = AngleMap(top_angle, bottom_angle, 1);
+
+    AngleSegment obstacle_angle_seg = AngleSegment(Angle::quarter(), Angle::zero());
+    angle_map.addNonViableAngleSegment(obstacle_angle_seg);
+
+    AngleSegment overlapping_angle_seg =
+        AngleSegment(Angle::fromDegrees(135), Angle::fromDegrees(45));
+    angle_map.addNonViableAngleSegment(overlapping_angle_seg);
+
+    EXPECT_EQ(45, angle_map.getBiggestViableAngleSegment().getDelta());
+}
+
+/**
+ * Example of undefined behaviour
+ * Two AngleSegment's not overlapping would be joined by the third ideally, but we
+ * don't define this behaviour. This scenario can be solved by sorting before hand
+ * instead
```

Comment:
is this comment still valid? can we uncomment the test now?

Reply:
Since it's undefined, I don't think we can/shouldn't really test it. It's nice to have an example though, I can remove if you want

Reply:
maybe im not fully understanding the comment here, what do we not support + where does sorting come in?

Reply:
if we have two angle segments:
a = 0 -> 45
b = 90 -> 180

these two get added to the map

now we have another angle segment:
c = 45 -> 90

if c gets added to the map it will only merge a or b, not both

the map will still have two angle segments even though it should only have one since they all overlap

sorting before hand solves this so they'll all be in an incremental order

Reply:
> if c gets added to the map it will only merge one of a or b, not both

is it always a or always b in this case? 

regardless this is a good explanation, we should add this to the angle map header and can probably remove this test

Reply:
angle map user should be aware inputs have to be sorted in counterclockwise order for the merging to happen properly (if thats true)

Reply:
> > if c gets added to the map it will only merge one of a or b, not both
> 
> is it always a or always b in this case?
> 
> regardless this is a good explanation, we should add this to the angle map header and can probably remove this test

it'll always be in the order of which one gets added first

Reply:
ya ill remove it and add this to the header

Reply:
done